omx/gstomxvideodec.c

```diff
@@ -1600,16 +1600,6 @@ gst_omx_video_dec_loop (GstOMXVideoDec * self)
   {
     g_mutex_lock (&self->drain_lock);
     if (self->draining) {
-      /* In reverse playback, we just need to confirm the component has processed
```

Comment:
So if we failed doing a buffer acquire due to an EOS, we originally jumped here and query a drain, that for forward playback only. But that apparently gets the pipe into trouble, right?

Reply:
Yes, this logic is exclusive of the draining process, a normal EOS from the acquire buffer doesn't trigger this query. Now that we activated the empty EOS buffer draining method for reverse playback, this query drain was being triggered and depending on the state of the other elements it was detected that this query can block the pipeline, specifically in some special cases when the stream position, the total duration, and the pipeline state change the conditions of the other elements.
Somehow this query drain can interfere with the normal seeking process and block the pipeline. In a decoder drain call we just need to ensure that the decoder component is empty and has processed all the received buffers, it shouldn't interfere with the drain process of other elements.